### ISSUE TITLE
FCM does not show the icon and app name in heads-up notification at Android 9+

### ISSUE THREAD

**Issue by taduuu** [invertase/react-native-firebase#3461]
**TL;DR:** 
Firebase cloud message does not show the icon of my application in "heads-up" notification only at Android 9+ devices. https://developer.android.com/guide/topics/ui/notifiers/notifications#Heads-up

**Longer version:**
When I send a cloud message directly from Firebase or by any Rest Client to the https://fcm.googleapis.com/fcm/send endpoint everything looks fine, the message shows up at the screen with a title, description and icon, device makes a sound, it disappears in a while but stays in the notification bar and notification drawer.

Everything works fine until the message is delivered to a device that runs on Android 10 or 9 (but not all of them). Notification bar and notification drawer works fine and the message itself looks exactly the same as in older Android versions. BUT the heads-up misses the icon of my application. Until now I have not been sending any "icon" with payload to Firebase, but even when I set the "icon" property with a URL to my icon, it still does not appear at the heads-up. I have put the same URL in an "image" property and then the image appears in the heads-up, but I believe it is not the property that should be used to show the application icon in the notification.

Has anybody noticed the same issue? I tried to look in the Google Android documentation - but it looks like out of date and ends with Android 8. I tried Firebase documentation but it says that "icon" property is optional (this is why I have not attached it).

I have tested notifications at:

Android device: 5.1(works), 7(works), 8.1(works), 9(works!), another 9(does not work), 10(does not work)
emulator of android: 8.1(works), 9(does not work), 10(does not work).
Payload is just:

```
{
    "to": "(...)",
    "notification": {
        "title": "Push from RestClient",
        "body": "Notification body"
    },
    "data": {}
}
```

**Of course I have added the required by Firebase for default_notification_icon, color and channel. This is not a solution here.**

**Comment by Ehesp** [comment]
This is out of our control, we don't control how notifications are created.

I'd take a good guess at saying your current icon resource (within the `res` directory) is setup for pre Android 9, whereas on new APIs it looks for a newer format of icon.

Best bet would be to open Android studio and add a new icon resource directly from there, since it'll create all formats for new and old device APIs.

Closing but feel free to fire any questions.
